client/my-sites/earn/ads/form-settings.jsx

```diff
@@ -478,71 +461,41 @@ class AdsFormSettings extends Component {
 		);
 	}
 
-	render() {
-		const { isLoading, site, translate } = this.props;
+	return (
+		<Fragment>
+			<QueryWordadsSettings siteId={ site.ID } />
 
-		const isWordAds = this.props.site.options.wordads;
+			<SectionHeader label={ translate( 'Ads Settings' ) }>
+				<Button compact primary onClick={ handleSubmit } disabled={ isLoading || ! isWordAds }>
+					{ isLoading ? translate( 'Saving…' ) : translate( 'Save Settings' ) }
+				</Button>
+			</SectionHeader>
 
-		return (
-			<Fragment>
-				<QueryWordadsSettings siteId={ site.ID } />
-
-				<SectionHeader label={ translate( 'Ads Settings' ) }>
-					<Button
-						compact
-						primary
-						onClick={ this.handleSubmit }
-						disabled={ isLoading || ! isWordAds }
-					>
-						{ isLoading ? translate( 'Saving…' ) : translate( 'Save Settings' ) }
-					</Button>
-				</SectionHeader>
-
-				<Card>
-					<form
-						id="wordads-settings"
-						onSubmit={ this.handleSubmit }
-						onChange={ this.props.markChanged }
-					>
-						{ this.showAdsToOptions() }
-
-						{ this.displayOptions() }
-
-						{ this.privacy() }
-
-						{ this.props.siteIsJetpack ? this.adstxt() : null }
-
-						<FormSectionHeading>{ translate( 'Payment Information' ) }</FormSectionHeading>
-						{ this.paymentOptions() }
-
-						<FormSectionHeading>{ translate( 'Terms of Service' ) }</FormSectionHeading>
-						{ this.acceptCheckbox() }
-					</form>
-				</Card>
-			</Fragment>
-		);
-	}
-}
-
-export default compose(
-	connect(
-		( state ) => {
-			const siteId = getSelectedSiteId( state );
-			const isJetpack = isJetpackSite( state, siteId );
-			const isSavingSettings = isSavingWordadsSettings( state, siteId );
-			const wordadsSettings = getWordadsSettings( state, siteId );
-
-			return {
-				isLoading: isSavingSettings || ! wordadsSettings,
-				site: getSelectedSite( state ),
-				siteIsJetpack: isJetpack,
-				wordadsSettings,
-				widgetsUrl: getCustomizerUrl( state, siteId, 'widgets' ),
-				siteUrl: getSiteUrl( state, siteId ),
-			};
-		},
-		{ dismissWordAdsSuccess, saveWordadsSettings }
-	),
-	localize,
-	protectForm
-)( AdsFormSettings );
+			<Card>
+				<form
+					id="wordads-settings"
+					onSubmit={ handleSubmit }
+					onChange={ () => setIsChanged( true ) }
+				>
+					<ProtectFormGuard isChanged={ isChanged } />
```

Comment:
The ProtectFormGuard component replaces the `protectForm` HOC on line 547 in the old code. This is a different implementation of the same logic.

As part of this change, the markChanged and markSaved methods (which were provided by the HOC) have been removed. They're replaced by the isChanged state. When any change happens on the form we run `setIsChanged( true )`. And in the handlSubmit method, we reset this to false with `setIsChanged( false )`.

You can see the [readme docs for this component](https://github.com/Automattic/wp-calypso/blob/trunk/client/lib/protect-form/README.md) for details.